znai-server/src/main/java/com/twosigma/znai/server/DocumentationServer.java

```diff
@@ -116,6 +132,21 @@ public HttpServer create() {
         return server;
     }
 
+    private String getActorFromQueryString(String queryString) {
```

Comment:
I feel like there should be a default way to get a query param. 
I haven't tried, but seems like there is a way to get a user from a principal instead of a query param
https://vertx.io/docs/apidocs/io/vertx/ext/auth/User.html#principal--

Reply:
I will see if I can adopt the principal.

Reply:
I look at the document about authentication. That's not what I intend to change. I want to pass an additional query parameter to pass an actor, not to support an authentication. For example, an authentication key for authentication might be the same as an actor.

Reply:
What I was trying to check with you is I think there is already user information stored inside principal value due to all your requests being kerberized on your side.

Reply:
I checked headers when a request hits the endpoint.
I couldn't find any header related to Kerberos.
I guess that as znai is not integrated with gss library, it doesn't reject a request with 407 to ask a kerberos ticket.

Reply:
Weird. Integration suppose to happen automatically as long as you built internally. It worked for webtau. I think as long as BaseInit.init is called and you use internal java to build it should work.

Reply:
I think that webtau is a client library, not the service. If a service rejects a request for Kerberos, the internal http client will fetch kerberos ticket from KDC and resend the request with the header for kerberos. If webtau hits a service which doesn't require Kerberos authentication, it won't ask Kerberos ticket from KDC.

Reply:
Ah yes you are right! Let’s proceed with query param. Apologies for the delay.

Reply:
But let’s use the standard parser for query params.

Reply:
I already changed the code to get an actor from ctx.request().params().